packages/typespec-client-generator-core/src/http.ts

```diff
@@ -655,7 +655,9 @@ function filterOutUselessPathParameters(
       param.__raw &&
       isPathParam(context.program, param.__raw) &&
       httpOperation.parameters.parameters.filter(
-        (p) => p.type === "path" && p.name === getWireName(context, param.__raw!),
+        (p) =>
+          p.type === "path" &&
+          p.name === (getPathParamName(context.program, param.__raw!) ?? param.name),
```

Comment:
we have to use `__raw`?

Reply:
does this mean we should filter by the parameter's name instead of its wire name, or the other way around?

Reply:
I think maybe we should check the part that never changes, here we should be checking if there wire name is the same, something like:
```
p.wireName === getWireName(context, param.__raw!)
```

Reply:
`getWireName` does not support path/query parameter. the goal here is to filter out method parameter which is a path parameter but not used in operation parameter list. so, i believe use query param name is reasonable.